bug: running make build and make serve fails

* On running `make build`
the following error comes up:
```
╰─ make build && make serve 
hugo --minify
Start building sites … 
Total in 1497 ms
Error: Error building site: POSTCSS: failed to transform "scss/main.css" (text/css): internal/modules/cjs/loader.js:888
  throw err;
  ^

Error: Cannot find module 'postcss'
Require stack:
- /usr/local/lib/node_modules/postcss-cli/index.js
- /usr/local/lib/node_modules/postcss-cli/bin/postcss
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:885:15)
    at Function.Module._load (internal/modules/cjs/loader.js:730:27)
    at Module.require (internal/modules/cjs/loader.js:957:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/postcss-cli/index.js:14:17)
    at Module._compile (internal/modules/cjs/loader.js:1068:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:933:32)
    at Function.Module._load (internal/modules/cjs/loader.js:774:14)
    at Module.require (internal/modules/cjs/loader.js:957:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/local/lib/node_modules/postcss-cli/index.js',
    '/usr/local/lib/node_modules/postcss-cli/bin/postcss'
  ]
}
```
Fix: run `npm install` for the first time you clone the project.

* On running `make serve`, the following error comes up:
```
╰─ make serve               
hugo server --buildFuture
Start building sites … 

                   | EN   
-------------------+------
  Pages            | 105  
  Paginator pages  |   0  
  Non-page files   |   1  
  Static files     |  48  
  Processed images |   2  
  Aliases          |   0  
  Sitemaps         |   1  
  Cleaned          |   0  

Built in 869 ms
Watching for changes in /Users/yashvardhankukreja/Projects/GoProjects/website/{assets,content,i18n,layouts,package.json,static,themes}
Watching for config changes in /Users/yashvardhankukreja/Projects/GoProjects/website/config/_default, /Users/yashvardhankukreja/Projects/GoProjects/website/config/_default/menus, /Users/yashvardhankukreja/Projects/GoProjects/website/themes/docsy/config.toml
fatal error: pipe failed

goroutine 1 [running]:
runtime.throw(0x5b2ab88, 0xb)
        runtime/panic.go:1117 +0x72 fp=0xc003497890 sp=0xc003497860 pc=0x403aad2
runtime.sigNoteSetup(0x6b7d180)
        runtime/os_darwin.go:98 +0xc5 fp=0xc0034978b8 sp=0xc003497890 pc=0x4036fc5
os/signal.signal_enable(0xb20bc15300000002)
        runtime/sigqueue.go:228 +0xa5 fp=0xc0034978d8 sp=0xc0034978b8 pc=0x4070e65
os/signal.enableSignal(...)
        os/signal/signal_unix.go:49
os/signal.Notify.func1(0x2)
        os/signal/signal.go:145 +0x88 fp=0xc0034978f8 sp=0xc0034978d8 pc=0x549ff68
os/signal.Notify(0xc001d9c000, 0xc003497b38, 0x2, 0x2)
        os/signal/signal.go:165 +0x185 fp=0xc003497970 sp=0xc0034978f8 pc=0x549f985
github.com/gohugoio/hugo/commands.(*commandeer).serve(0xc000526e00, 0xc00037e5c0, 0xbd, 0x100)
        github.com/gohugoio/hugo/commands/server.go:494 +0x629 fp=0xc003497b68 sp=0xc003497970 pc=0x54d3ca9
github.com/gohugoio/hugo/commands.(*serverCmd).server(0xc00037e5c0, 0xc0004ff900, 0xc000247770, 0x0, 0x1, 0x0, 0x0)
        github.com/gohugoio/hugo/commands/server.go:272 +0x2c5 fp=0xc003497cb0 sp=0xc003497b68 pc=0x54d2465
github.com/gohugoio/hugo/commands.(*serverCmd).server-fm(0xc0004ff900, 0xc000247770, 0x0, 0x1, 0x0, 0x0)
        github.com/gohugoio/hugo/commands/server.go:131 +0x5b fp=0xc003497cf8 sp=0xc003497cb0 pc=0x54e071b
github.com/spf13/cobra.(*Command).execute(0xc0004ff900, 0xc000247760, 0x1, 0x1, 0xc0004ff900, 0xc000247760)
        github.com/spf13/cobra@v1.1.3/command.go:852 +0x472 fp=0xc003497db8 sp=0xc003497cf8 pc=0x41d5712
github.com/spf13/cobra.(*Command).ExecuteC(0xc000530280, 0xc0002475e0, 0x5, 0x6)
        github.com/spf13/cobra@v1.1.3/command.go:960 +0x375 fp=0xc003497e98 sp=0xc003497db8 pc=0x41d6295
github.com/gohugoio/hugo/commands.Execute(0xc0000ae0a0, 0x2, 0x2, 0x4009265, 0xc0000a0058, 0x54e4630, 0x0)
        github.com/gohugoio/hugo/commands/hugo.go:87 +0xb9 fp=0xc003497f28 sp=0xc003497e98 pc=0x54bee79
main.main()
        github.com/gohugoio/hugo/main.go:23 +0x76 fp=0xc003497f88 sp=0xc003497f28 pc=0x54e4016
runtime.main()
        runtime/proc.go:225 +0x256 fp=0xc003497fe0 sp=0xc003497f88 pc=0x403d2f6
runtime.goexit()
        runtime/asm_amd64.s:1371 +0x1 fp=0xc003497fe8 sp=0xc003497fe0 pc=0x4074c21

goroutine 21 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc0001c6180)
        go.opencensus.io@v0.23.0/stats/view/worker.go:276 +0xcd
created by go.opencensus.io/stats/view.init.0
        go.opencensus.io@v0.23.0/stats/view/worker.go:34 +0x68

goroutine 111 [syscall]:
syscall.syscall6(0x424f820, 0x7, 0x0, 0x0, 0xc0005efe88, 0xa, 0x6b7cc40, 0x0, 0x0, 0x0)
        runtime/sys_darwin.go:41 +0x2e
golang.org/x/sys/unix.kevent(0x7, 0x0, 0x0, 0xc0005efe88, 0xa, 0x6b7cc40, 0x0, 0x0, 0x0)
        golang.org/x/sys@v0.0.0-20210412220455-f1c623a9e750/unix/zsyscall_darwin_amd64.go:276 +0xa5
golang.org/x/sys/unix.Kevent(0x7, 0x0, 0x0, 0x0, 0xc0005efe88, 0xa, 0xa, 0x6b7cc40, 0x0, 0x0, ...)
        golang.org/x/sys@v0.0.0-20210412220455-f1c623a9e750/unix/syscall_bsd.go:429 +0x71
github.com/fsnotify/fsnotify.read(0x7, 0xc0005efe88, 0xa, 0xa, 0x6b7cc40, 0xc0005efe88, 0x0, 0xa, 0x0, 0x0)
        github.com/fsnotify/fsnotify@v1.4.9/kqueue.go:511 +0x6e
github.com/fsnotify/fsnotify.(*Watcher).readEvents(0xc001ca2300)
        github.com/fsnotify/fsnotify@v1.4.9/kqueue.go:274 +0x81b
created by github.com/fsnotify/fsnotify.NewWatcher
        github.com/fsnotify/fsnotify@v1.4.9/kqueue.go:62 +0x199

goroutine 112 [select]:
github.com/gohugoio/hugo/watcher.(*Batcher).run(0xc0024dbfe0)
        github.com/gohugoio/hugo/watcher/batcher.go:53 +0x129
created by github.com/gohugoio/hugo/watcher.New
        github.com/gohugoio/hugo/watcher/batcher.go:42 +0x13b

goroutine 113 [select]:
github.com/gohugoio/hugo/commands.(*commandeer).newWatcher.func1(0xc0024dbfe0, 0xc000526e00, 0xc00063cc80, 0xc00441c9f0)
        github.com/gohugoio/hugo/commands/hugo.go:853 +0xc5
created by github.com/gohugoio/hugo/commands.(*commandeer).newWatcher
        github.com/gohugoio/hugo/commands/hugo.go:851 +0x2ce

goroutine 216 [runnable]:
github.com/gohugoio/hugo/livereload.(*hub).run(0x6b3cd40)
        github.com/gohugoio/hugo/livereload/hub.go:37
created by github.com/gohugoio/hugo/livereload.Initialize
        github.com/gohugoio/hugo/livereload/livereload.go:99 +0x45
make: *** [serve] Error 2
```

Fix: Run `hugo server --buildFuture --watch=false` instead to start the hugo server healthily

--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,16 @@ CCEND=\033[0m
 help: ## Show this help.
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {sub("\\\\n",sprintf("\n%22c"," "), $$2);printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-module-check:
+npm-dependencies-install: ## Install the required npm dependencies backing the website, only if they haven't been installed yet
+	@if [ ! -d "node_modules" ]; then npm install; else echo "npm dependencies already found to be installed. To re-install, run make npm-dependencies-force-install"; fi
+
+npm-dependencies-purge: ## Purge all the npm dependencies installed under this project
+	@rm -rf node_modules
+
+npm-dependencies-force-install: ## Install the required npm dependencies backing the website even if they already were installed
+	@npm install
+
+module-check: npm-dependencies-install
 	@git submodule status --recursive | awk '/^[+-]/ {printf "\033[31mWARNING\033[0m Submodule not initialized: \033[34m%s\033[0m\n",$$2}' 1>&2
 
 all: build ## Build site with production settings and put deliverables in ./public
@@ -44,7 +53,7 @@ non-production-build: ## Build the non-production site, which adds noindex heade
 	hugo --enableGitInfo
 
 serve: module-check ## Boot the development server.
-	hugo server --buildFuture
+	hugo server --buildFuture --watch=false
 
 container-image:
 	$(CONTAINER_ENGINE) build . \

